File: liqp-core/src/main/java/liqp/filters/First.java
package liqp.filters;

import liqp.nodes.RenderContext;

class First extends Filter {

    /*
     * first(array)
     *
     * Get the first element of the passed in array
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        Object[] array = super.asArray(value);

        return array.length == 0 ? null : super.asString(array[0]);
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/First.kt
package liqp.filter

import liqp.context.LContext

class First : LFilter() {

  /**
   * first(array)
   *
   * Get the first element of the passed in array
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asIterable(value).firstOrNull()
  }
}
